sandbox/modules/identity/src/main/java/org/opensearch/identity/rest/validation/CredentialsValidator.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.rest.validation;
+
+import org.opensearch.common.Strings;
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.compress.NotXContentException;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.XContentHelper;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.identity.ConfigConstants;
+import org.opensearch.identity.utils.Utils;
+import org.opensearch.rest.RestRequest;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * Validator for validating password and hash present in the payload
+ */
+public class CredentialsValidator extends AbstractConfigurationValidator {
+
+    public CredentialsValidator(final RestRequest request, BytesReference ref, final Settings opensearchSettings, Object... param) {
+        super(request, ref, opensearchSettings, param);
+        this.payloadMandatory = true;
+        allowedKeys.put("hash", DataType.STRING);
+        allowedKeys.put("password", DataType.STRING);
+    }
+
+    /**
+     * Function to validate password in the content body.
+     * @return true if validation is successful else false
+     */
+    @Override
+    public boolean validate() {
+        if (!super.validate()) {
+            return false;
+        }
+
+        final String regex = this.opensearchSettings.get(ConfigConstants.IDENTITY_RESTAPI_PASSWORD_VALIDATION_REGEX, null);
```

Comment:
wdyt about providing a default? Does it make sense to have this be configurable?